ui/pages/token-details/token-details-page.js

```diff
@@ -42,7 +42,9 @@ export default function TokenDetailsPage() {
 
   const { asset: tokenAddress } = assetAddress;
 
-  const tokenMetadata = tokenList[tokenAddress];
+  const tokenMetadata = Object.values(tokenList).find((token) =>
```

Comment:
I think it may be safe to alternatively do:

```
const tokenMetadataKey = useTokenDetection ? tokenAddress.toLowerCase() : tokenAddress;
const tokenMetadata = tokenList[tokenMetadataKey];
```

Reply:
if we knew for certain that `tokenList` only had lowercase addresses, and if we are not concerned about `useTokenDetection` for this logic, then I think we could use 
```
const tokenMetadata = tokenList[tokenAddress?.toLowerCase()];
```

This said I think the implementation you created in this PR is more robust since it doesn't involve assuming, or ensuring, `tokenList` addresses are also lowercase.